src/lib/fixed_string.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+// FixedString is a data type, which stores a string in an array of chars in order to
+// save memory space by avoiding SSO
+class FixedString {
+ public:
+  // Create a FixedString from a std::string
+  // We did not find any good solution for creating a char array on stack
+  explicit FixedString(const std::string& string) : _mem(new char[string.size()]{}), _string_length(string.size()) {
+    std::memcpy(_mem, string.c_str(), _string_length);
+  }
+
+  // Create a FixedString from a memory address
+  FixedString(char* mem, size_t string_length) : _mem(mem), _string_length(string_length), _delete(false) {}
+
+  FixedString(FixedString& other) : _mem(new char[other._string_length]{}), _string_length(other._string_length) {
+    std::memcpy(_mem, other._mem, _string_length);
+  }
+
+  // Create a FixedString with an existing one
+  FixedString(const FixedString& other) : _mem(new char[other._string_length]{}), _string_length(other._string_length) {
+    std::memcpy(_mem, other._mem, _string_length);
+  }
+
+  FixedString(const FixedString&& other)
+      : _mem(new char[other._string_length]{}), _string_length(other._string_length) {
+    std::memcpy(_mem, other._mem, _string_length);
+  }
```

Comment:
We cannot change the members _mem, _string_length and _delete from the "other" object since they are const. And we would like to keep them const because we create many FixedString objects when using iterators on the ValueVector.

Bending the _mem pointer to the other char array would be good. We are not sure what would happen if the destructor from the "other" object is called.